statusHandler/statusMetricsProvider.go

```diff
@@ -97,40 +116,46 @@ func (sm *statusMetrics) Close() {
 
 // StatusMetricsMapWithoutP2P will return the non-p2p metrics in a map
 func (sm *statusMetrics) StatusMetricsMapWithoutP2P() map[string]interface{} {
+	sm.mutOperations.RLock()
+	defer sm.mutOperations.RUnlock()
+
 	statusMetricsMap := make(map[string]interface{})
-	sm.nodeMetrics.Range(func(key, value interface{}) bool {
-		keyString := key.(string)
-		if strings.Contains(keyString, "_p2p_") {
-			return true
+	for key, value := range sm.nodeMetrics {
+		if strings.Contains(key, "_p2p_") {
+			continue
 		}
 
-		statusMetricsMap[key.(string)] = value
-		return true
-	})
+		statusMetricsMap[key] = value
+	}
 
 	return statusMetricsMap
 }
 
 // StatusP2pMetricsMap will return the p2p metrics in a map
 func (sm *statusMetrics) StatusP2pMetricsMap() map[string]interface{} {
+	sm.mutOperations.RLock()
+	defer sm.mutOperations.RUnlock()
+
 	statusMetricsMap := make(map[string]interface{})
-	sm.nodeMetrics.Range(func(key, value interface{}) bool {
-		keyString := key.(string)
-		if !strings.Contains(keyString, "_p2p_") {
-			return true
+	for key, value := range sm.nodeMetrics {
+		if strings.Contains(key, "_p2p_") {
+			continue
 		}
 
-		statusMetricsMap[key.(string)] = value
-		return true
-	})
+		statusMetricsMap[key] = value
+	}
 
 	return statusMetricsMap
 }
 
 // StatusMetricsWithoutP2PPrometheusString returns the metrics in a string format which respects prometheus style
 func (sm *statusMetrics) StatusMetricsWithoutP2PPrometheusString() string {
```

Comment:
do we still need this?

Reply:
yes, we cannot tell is anyone uses this endpoint. since it's used for prometheus exported metrics, there are chances that a few people monitor their node by using this